specification/sbi_atee.adoc

```diff
@@ -1490,86 +1574,131 @@ struct AttestationCapabilities {
+struct sbiret sbi_tee_guest_get_evidence(unsigned long pub_key_addr,
+                                         unsigned long pub_key_size,
+                                         unsigned long challenge_data_addr,
+                                         unsigned long cert_format,
+                                         unsigned long cert_addr_out,
+                                         unsigned long cert_size);
 -------
-If the `sbi_tee_guest_get_attcaps` enumerates attestation services provided by the TSM, then 
-this intrinsic is used by a TVM to get attestation evidence to report to a (remote) relying party. 
-This may take the form of a request for an attestation certificate or a TSM-signed TVM 
-measurement (using an attestation certificate specific to the TVM).
-
-Get attestation evidence from a Certificate Signing Request (CSR) 
-per https://datatracker.ietf.org/doc/html/rfc2986. The caller passes the CSR and its length 
-through the first 2 arguments. The third argument is the address where the caller 
-places a data blob that will be included in the generated certificate. 
-Typically, this is a cryptographic nonce. The fourth argument is the evidence 
-format: DiceTcbInfo (0), DiceMultiTcbInfo (1) or OpenDice (2). The fifth argument 
-is the address where the generated certificate will be placed. The evidence is 
-formatted an x.509 DiceTcbInfo certificate extension
-
-It is supported by the TSM to provide HW-key-signed measurements of the TVM and the TSM. 
-The attestation key used to sign the evidence is provisioned into the TVM by the TSM. 
-The TSM certificate is provisioned by the FW TCB (TSM-driver and HW RoT).
-
-Both `cert_request_addr`, `request_data_addr` and `cert_addr_out` must be 4kB-aligned.
+If the `sbi_tee_guest_get_attcaps` enumerates attestation services provided by
+the TSM, then this intrinsic is used by a TVM to get an attestation evidence to
+report to a remote relying party.
+
+This intrisic returns an attestation certificate at the address passed as its
+fifth argument (`cert_addr_out`). The certificate is signed by the TSM
+attestation key, and includes the TVM attestation evidence. The TSM attestion
+key is also included in the reported TSM token.
+
+The caller passes the TVM public key address as the first argument
+(`pub_key_addr`). This key will be included in the generated certificate and
+represents the TSM-certified TVM identity.
+
+The third argument (`challenge_data_addr`) points to the attestation challenge
+blob, typically a relying party generated nonce used for demonstrating the
+attestation evidence fresheness.
+
+The fourth argument (`cert_format`) is the caller's selected attestation
+certificate format. This must be one of the supported `ATTESTATION_CERTIFICATE_*`
+flag, per the attestation capabilities reported via `sbi_tee_guest_get_attcaps`.
+
+All addresses (`pub_key_addr`, `request_data_addr` and `cert_addr_out`) must be
+4kB-aligned.
```

Comment:
do we qualify these addresses to be in confidential memory regions? we should IMO.

Reply:
`pub_key_addr` and `challenge_data_addr` definitely should. `cert_addr_out` is signed and does not contain any confidential information, so it does not need to be in confidential memory?

Reply:
yeah makes sense.